File: code/4.cc
#include <iostream>
#include <cmath>
#include <algorithm>
#include <string>
using namespace std;

bool ispal(int& i);

int main(){
	int num1 = 100;
	int largest = 0;
	while(num1 < 1000){
		int num2 = 100;
		while(num2 < 1000){
			int num3 = num1*num2;
			if(ispal(num3) && num3 > largest)
				largest = num3;
			num2++;
		}
		num1++;
	}
	cout<<largest<<endl;
	
	return 0;
}

bool ispal(int& i){
	string s = to_string(i);
	string s2 = s;
	reverse(s2.begin(), s2.end());
	if(s == s2) return true;
	else return false;
}

File: code/4.py
def ispal(i):
	s = str(i)
	s2 = s[::-1]
	if s == s2:
		return 1
	else:
		return 0

num1 = 100
largest = 0
while num1 < 1000:
	num2 = 100
	while num2 < 1000:
		num3 = num1*num2
		if ispal(num3) == 1 and num3 > largest:
			largest = num3
		num2 += 1
	num1 += 1
print(largest)
